### ISSUE TITLE
Bug in ether compiler

### ISSUE THREAD

**Issue by ghost** [SBasalaev/alchemy-os#5]
code:

```
 var i: Int
 if ((i = in.read()) != -1) {
  ...
 }
```

generate IllegalStateException in time compiling. Message: "Stack inconsistency: empty stack"

**Comment by SBasalaev** [comment]
Thanks for reporting that, will be fixed in the next bugfix release.
Actually, in Ether that kind of code is illegal, assignment never
returns a value, so you have to do it sequential way:

```
var i = in.read()
if (i != EOF) {
  ...
}
```

**Comment by SBasalaev** [comment]
Fixed in release 2.1.8
